### PR TITLE
Main structural changes, and some fixes

### PR DESCRIPTION
**Pull request by Mefjuu94**
- Splitted Service class to smallest parts (Handlers) to better maintain and clearest view code. each class is responsible for specific information.

- fixed responses, now matches as expected examples from api.
- Added new Package, with Tests classes Handlers with use Mockito Libray

### CODE REVIEW

File: src/main/java/org/example/ApiServiceConnections/ApiConnectionConstans.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package org.example.ApiServiceConnections;
+
+public class ApiConnectionConstans {
+    protected static final String URL = "https://www.dnd5eapi.co/api/classes/";
+
+    public static final String[] classes = {"barbarian", "bard", "cleric", "druid", "fighter",
```
Comment (blazejprzyluski) [inline src/main/java/org/example/ApiServiceConnections/ApiConnectionConstans.java]: Please put the variable name in upper case

File: src/main/java/org/example/ApiServiceConnections/ApiConnectionConstans.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+            "monk", "paladin", "ranger", "rogue", "sorcerer", "warlock", "wizard"};
+    protected static final String[] races = {"dragonborn", "dwarf", "elf", "gnome", "half-elf",
```
Comment (blazejprzyluski) [inline src/main/java/org/example/ApiServiceConnections/ApiConnectionConstans.java]: Please put the variable name in upper case

File: src/main/java/org/example/ApiServiceConnections/ApiConnectionConstans.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package org.example.ApiServiceConnections;
+
+public class ApiConnectionConstans {
```
Comment (blazejprzyluski) [inline src/main/java/org/example/ApiServiceConnections/ApiConnectionConstans.java]: ApiConnectionConstants*

File: src/main/java/org/example/ApiServiceConnections/ClassFeaturesHanlder.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+import org.example.Main;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+
+public class ClassFeaturesHanlder {
```
Comment (blazejprzyluski) [inline src/main/java/org/example/ApiServiceConnections/ClassFeaturesHanlder.java]: Handler*

File: src/test/java/ApiHandlersConnectionMockitoTests.java [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+import java.net.http.HttpResponse;
+
+import static org.mockito.Mockito.lenient;
+import static org.mockito.Mockito.when;
+
+
+@ExtendWith(MockitoExtension.class)
+public class ApiHandlersConnectionMockitoTests {
+
+    private final String URL = "https://www.dnd5eapi.co/api/classes/";
```
Comment (blazejprzyluski) [inline src/test/java/ApiHandlersConnectionMockitoTests.java]: isn't this already available in the Constants class?

File: src/test/java/ApiHandlersConnectionMockitoTests.java [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+
+    private final String[] classes = {"barbarian", "bard", "cleric", "druid", "fighter",
```
Comment (blazejprzyluski) [inline src/test/java/ApiHandlersConnectionMockitoTests.java]: same here

File: src/main/java/org/example/ApiServiceConnections/ApiConnectionConstans.java (hunk 1 above)
Comment (Mefjuu94) [reply src/main/java/org/example/ApiServiceConnections/ApiConnectionConstans.java]: Done!

File: src/main/java/org/example/ApiServiceConnections/ApiConnectionConstans.java (hunk 2 above)
Comment (Mefjuu94) [reply src/main/java/org/example/ApiServiceConnections/ApiConnectionConstans.java]: Done!

File: src/main/java/org/example/ApiServiceConnections/ApiConnectionConstans.java (hunk 3 above)
Comment (Mefjuu94) [reply src/main/java/org/example/ApiServiceConnections/ApiConnectionConstans.java]: ok!

File: src/main/java/org/example/ApiServiceConnections/ClassFeaturesHanlder.java (hunk 4 above)
Comment (Mefjuu94) [reply src/main/java/org/example/ApiServiceConnections/ClassFeaturesHanlder.java]: done!

File: src/test/java/ApiHandlersConnectionMockitoTests.java (hunk 5 above)
Comment (Mefjuu94) [reply src/test/java/ApiHandlersConnectionMockitoTests.java]: ok

File: src/test/java/ApiHandlersConnectionMockitoTests.java (hunk 6 above)
Comment (Mefjuu94) [reply src/test/java/ApiHandlersConnectionMockitoTests.java]: ok!

**Pull request merged**